nft-rfc-006.md

```diff
@@ -0,0 +1,860 @@
+architecture that allows for decentralized and future proofed innovation.
+
+Third, DID Methods with publicly inspectable registries (where DID
+Document state is maintained), such as blockchains, may also provide for
+auditability, and with some effort, even reversibility of changes in
+case of inappropriate transactions. These advanced features remain areas
+of active development.
+
+Finally, DID Documents are defined using an abstract data model,
+allowing any number of serializations, with initial support for JSON and
+JSON-LD serializations. CBOR and CBOR-LD are also highly anticipated for
+their ability to dramatically reduce the size of DID Documents. Other,
+future serializations are also possible without losing conformance to the specification.
+
+### Lessons Learned from DIDs
+
+First and foremost, do not use public ledgers as directories. Only use
+them for verification of temporal ordering. That is, with a public
+ledger, it is possible to deterministically know which DID Document is
+the current, authentic DID Document for a given DID. Some Methods do not
+provide this verifiability, which may be fine for some use cases. Using
+public ledgers as directories presents privacy and confidentiality
+problems as ledgers are designed to immutably record their transactions:
+once written to the ledger, you generally can\'t unwrite it, making
+disclosure of proprietary or personal information problematic.
+
+Second, avoid unnecessary public correlation. DID Documents provide the
+option of embedding service endpoints, which may themselves be present
+in multiple DID Documents. For example, you \*could\* add a Twitter
+address as a service endpoint in several different DIDs. This would link
+those DIDs to each other in a non-verifiable yet undisputable way: while
+observers could not directly tell if those service endpoints mean that
+the Twitter account controls those DIDs, they do gain a point of
+observation for consideration in evaluating that correlation. For many,
+this undermines the privacy goals of being able to use DIDs without
+revealing any additional personally identifiable information.
+
+Third, DIDs, and their sister specification VCs, embrace an open world
+data model, in which anyone can add any parameter to the Document
+without violating conformance. This is convenient, but also means that
+unfortunate properties can and will be used, such as properties that
+explicitly communicate details about the DID Subject. Such details are
+intended to be preserved by Methods and requesting parties even when
+they are misunderstood, which creates significant opportunity for
+unintended consequences. However, it is understood that this flexibility
+affords significant innovation even as it invites endless debate on
+which properties get to be defined where---a decision that has
+significant impact on ubiquitous adoption.
+
+Finally, DIDs are NOT signed documents. The only verification afforded
```

Comment:
Is it possible to sign a did document using the `Proof` section of a DID document? Maybe I'm misunderstanding what a `signed document` is in this context?

https://www.w3.org/2019/08/did-20190828/#proof-optional

Reply:
Actually, the comment above is linked to an older version of the w3c spec, I think the most recent version has no concept of `proofs` on the top-level data structure but it does have a way to prove authenticity 

https://w3c.github.io/did-core/#proving-control-of-a-did-and-did-document

Reply:
It would absolutely be possible to sign a DID Document, using JSON-LD Signatures. Any JSON-LD document can be signed in that way.

The reason DID Documents are not signed is because some early DID Methods, like BTCR, don't know the DID itself--the identifier string that is the DID--until the DID is on-chain, and putting the DID on-chain included adding a hash (typically an IPFS address) of the DID Document. But the DID Document MUST have the DID in it. This leads to a tautological problem, because you can't sign the document without the DID, but you can't get the DID until after you register it.

That said, DID Methods whose identifiers are NOT cryptographically tied to the content of the DID Document *can* be signed.

AND, DID Methods whose identifiers ARE cryptographically tied to the content of the DID Document, e.g., via a hash or the document embedded in the DID itself, do not need to be signed (the proof is in the identifier).

Which of these is true for IIDs is likely to be namespace- (or method-) specific. If your IID Method has a way to construct the IID before it is on-chain, then you can sign the IID Document. But, like BTCR, there may not be obvious ways to do this for all IIDs.

There is also a class of DID whose identifier includes a hash of the *initial* DID Document, but which updates to that document are stored on-chain. That allows sophisticated DID Documents to be securely identified within the DID (all off-chain) while also allowing on-chain updates.

In the most privacy-respecting, anti-censorship pattern, one would construct an DID that embeds a hash of the initial DID Document, which is delivered independently of the DID--meaning that the DID Document is never publicly revealed. Then, you record a new hash for updates--again keeping the same DID and keeping the DID Document private. Only those with access or permission to access the current DID Document can see it, and yet, the chain ensures the ability to verify that any given DID Document is the current, authoritative one.

This is the approach being considered by some of the authors of BTCR for the next iteration on that BTC based DID.